pkg/block/azure/adapter.go

```diff
@@ -29,16 +32,26 @@ const (
 	// more the 5000 different accounts, which is highly unlikely
 	udcCacheSize = 5000
 
-	BlobEndpointGlobalFormat     = "https://%s.blob.core.windows.net/"
-	BlobEndpointChinaCloudFormat = "https://%s.blob.core.chinacloudapi.cn/"
+	BlobEndpointDefaultDomain = "blob.core.windows.net"
+)
+
+var (
+	ErrInvalidDomain = errors.New("invalid Azure Domain")
+
+	endpointRegex      = regexp.MustCompile(`https://(?P<account>[\w]+).(?P<domain>[\w.-]+)[/:][\w-/]*$`)
```

Comment:
Now you have 2 problems.....

Reply:
Not sure I understand your comment

Reply:
https://blog.codinghorror.com/regular-expressions-now-you-have-two-problems/

Reply:
Sometimes you just have to do it :)